lib/medtronic/medtronicSimulator.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+var _ = require('lodash');
+var sundial = require('sundial');
+var util = require('util');
+
+var annotate = require('../eventAnnotations');
+
+var debug = require('../bows')('MedtronicDriver');
+
+/**
+ * Creates a new "simulator" for Medtronic pump data.
+ *
+ * Once all input activities are collected, the simulator will have accumulated events that the Tidepool Platform
+ * will understand into a local "events" array.  You can retrieve the events by calling `getEvents()`
+ *
+ */
+exports.make = function(config){
+  var events = [];
+
+  var currBolus = null;
+  var currWizard = null;
+  var currTimestamp = null;
+
+  function setCurrBolus(bolus) {
+    currBolus = bolus;
+  }
+
+  function setCurrWizard(wizard) {
+    currWizard = wizard;
+  }
+
+  function ensureTimestamp(e){
+    if (currTimestamp > e.time) {
+      throw new Error(
+        util.format('Timestamps must be in order.  Current timestamp was[%s], but got[%j]', currTimestamp, e)
+      );
+    }
+    currTimestamp = e.time;
+    return e;
+  }
+
+  function simpleSimulate(e) {
+    ensureTimestamp(e);
+    events.push(e);
+  }
+
+  return {
+    bolus: function(event) {
+      if(currWizard && currWizard.jsDate.getTime() === event.jsDate.getTime()) {
```

Comment:
I just `git grep`ed and I believe these are the only instances of use of `getTime()`. I dislike the method due to the imprecision of its name (almost makes it sound like it's only getting the clock hours or something...). I would be in favor of `jsDate.valueOf()` instead - we use this a bit more and it returns the same value (a hammertime).